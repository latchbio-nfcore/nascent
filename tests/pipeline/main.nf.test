nextflow_pipeline {

    name "Test Workflow main.nf"
    script "main.nf"

    test("Should run without failures with bwa (default)") {

        when {
            params {
                outdir = "$outputDir"
                // aligner = "bwa"
            }
        }

        then {
            assert workflow.success
            assert workflow.trace.tasks().size() == 133
            assert snapshot(
                path("$outputDir/bbmap").list(),
                path("$outputDir/bed2saf").list(),
                path("$outputDir/bedtools").list(),
                // FIXME gtf2bed isn't deterministic
                // path("$outputDir/bwa/featurecounts/gene").list(),
                path("$outputDir/bwa/featurecounts/predicted").list(),
                // path("$outputDir/deeptools").list(),
                // path("$outputDir/fastp/*.json").list(),
                // FIXME gtf2bed isn't deterministic
                // path("$outputDir/gtf2bed").list(),
                path("$outputDir/homer/cd4.bed"),
                path("$outputDir/homer/jurkat.bed"),
                path("$outputDir/preseq").list(),
                // path("$outputDir/rseqc").list(),
                path("$outputDir/samtools").list(),
            ).match()
        }

    }

}
