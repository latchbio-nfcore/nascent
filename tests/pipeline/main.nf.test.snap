{
    "Should run with bwa (default)": {
        "content": [
            138,
            [
                "cd4_REP1.coverage.hist.txt:md5,7cbb473be8d3b32ff2e52fdf4e5d10d2",
                "cd4_REP1.coverage.stats.txt:md5,f1471b61ac17dba283d80e08450c7e55",
                "cd4_REP2.coverage.hist.txt:md5,8d5258a0882494bc4e3f1aa6aa5ed685",
                "cd4_REP2.coverage.stats.txt:md5,3cdb4473211f9da44166ffa6aaa5b602",
                "cd4_REP3.coverage.hist.txt:md5,4a8397c1ea08f35ca7046e3c2c014a08",
                "cd4_REP3.coverage.stats.txt:md5,b88592b469d504a93f2389255ac15f38",
                "cd4_REP4.coverage.hist.txt:md5,d2cbf89e1f753d99bcd03b4e472270dd",
                "cd4_REP4.coverage.stats.txt:md5,4c2b5fe6793aab8560cc87c176ac06ab",
                "jurkat_REP1.coverage.hist.txt:md5,71a893d9d1d55fd47399f6d47b628d6e",
                "jurkat_REP1.coverage.stats.txt:md5,381c69a30099d82066a959deab1a2569",
                "jurkat_REP2.coverage.hist.txt:md5,d124bdadaf2dc5cdd9e3db5a1ab4c588",
                "jurkat_REP2.coverage.stats.txt:md5,b1940346719e4070091f2eef6e508f49"
            ],
            [
                "cd4_intersect.saf:md5,149347636622d571c3c3ef762c1698ea"
            ],
            "cd4.bed:md5,ab94918610a560772fdbec591200295f",
            "jurkat.bed:md5,41908723fa423a78476feb2e7627a3a4",
            [
                "cd4_REP1.c_curve.txt:md5,cf4743abdd355595d6ec1fb3f38e66e5",
                "cd4_REP1.command.log:md5,d41d8cd98f00b204e9800998ecf8427e",
                "cd4_REP1.lc_extrap.txt:md5,6d0e7f782f34ee54213c022cbe19df5b",
                "cd4_REP2.c_curve.txt:md5,cf4743abdd355595d6ec1fb3f38e66e5",
                "cd4_REP2.command.log:md5,d41d8cd98f00b204e9800998ecf8427e",
                "cd4_REP2.lc_extrap.txt:md5,6aac6dd78dfd85a18fadd20be7377019",
                "cd4_REP3.c_curve.txt:md5,cf4743abdd355595d6ec1fb3f38e66e5",
                "cd4_REP3.command.log:md5,d41d8cd98f00b204e9800998ecf8427e",
                "cd4_REP3.lc_extrap.txt:md5,02c4b0fc0afd282a402a316d44598173",
                "cd4_REP4.c_curve.txt:md5,cf4743abdd355595d6ec1fb3f38e66e5",
                "cd4_REP4.command.log:md5,d41d8cd98f00b204e9800998ecf8427e",
                "cd4_REP4.lc_extrap.txt:md5,9efbab9d2aee791e7e9cae9237f2c500",
                "jurkat_REP1.c_curve.txt:md5,cf4743abdd355595d6ec1fb3f38e66e5",
                "jurkat_REP1.command.log:md5,d41d8cd98f00b204e9800998ecf8427e",
                "jurkat_REP1.lc_extrap.txt:md5,390386e428ce783b3a373a5f60ee912f",
                "jurkat_REP2.c_curve.txt:md5,cf4743abdd355595d6ec1fb3f38e66e5",
                "jurkat_REP2.command.log:md5,d41d8cd98f00b204e9800998ecf8427e",
                "jurkat_REP2.lc_extrap.txt:md5,cfdda91d83dd381dcefdc754e4b4c54c"
            ],
            [
                "GRCh38_chr21.fa.fai:md5,6ea51e1c1231da5e49212ea9f493d703",
                "cd4_REP1.flagstat:md5,863e2d506d5cc4239af98a5f31bbc906",
                "cd4_REP1.idxstats:md5,b1dd8bcbd23c53c21f0e11082d9315f2",
                "cd4_REP1.sorted.bam:md5,22e9e4ede572d35363dc3ec23eb9c79b",
                "cd4_REP1.sorted.bam.bai:md5,c5e37e318050b43fb6692cd86491d4b0",
                "cd4_REP1.stats:md5,f4aa5b8d3fea2797724bf2fa12961036",
                "cd4_REP2.flagstat:md5,0fd86dbf8f799fad49ba471702979bdc",
                "cd4_REP2.idxstats:md5,53204e4c6a9f68664087e4a8123be46a",
                "cd4_REP2.sorted.bam:md5,fb718ea76a1e22bda674e23a92662531",
                "cd4_REP2.sorted.bam.bai:md5,7f9d67f8572c50bddc57f8296b811fd8",
                "cd4_REP2.stats:md5,9952636336505197c44c9ce1f7314096",
                "cd4_REP3.flagstat:md5,793c9cadbf80e35dbae3774e9b5bd93f",
                "cd4_REP3.idxstats:md5,42de227ff397013cf288759871b0e938",
                "cd4_REP3.sorted.bam:md5,6043c68b66e294d4be039bd6c8390ed5",
                "cd4_REP3.sorted.bam.bai:md5,bb66de0f0efc0231d8252754cf8dc5c5",
                "cd4_REP3.stats:md5,4f63629b6850326e1dbf449732bc7bf3",
                "cd4_REP4.flagstat:md5,ed6c609505097f3cc9d9902d69e016f8",
                "cd4_REP4.idxstats:md5,959dc185ae59de4cebe374026711a55e",
                "cd4_REP4.sorted.bam:md5,d1d055ab384159184f6abf2b23cf8c9e",
                "cd4_REP4.sorted.bam.bai:md5,c70457bfeb054ab60e673aee4a247331",
                "cd4_REP4.stats:md5,87c41cda4168ed794b640bfbd0e99d45",
                "jurkat_REP1.flagstat:md5,fd5f02b0f02a407447b804b1d80f5421",
                "jurkat_REP1.idxstats:md5,c61af0847c1ad76c06a8de2815975b32",
                "jurkat_REP1.sorted.bam:md5,558e3948f1d263113d86f13dadead6a4",
                "jurkat_REP1.sorted.bam.bai:md5,62956925d0600af04cd40dc1c57fd110",
                "jurkat_REP1.stats:md5,677f4ec974fed98323ec41ab2f96049d",
                "jurkat_REP2.flagstat:md5,86ed47bd41a745ab59de473082c7742d",
                "jurkat_REP2.idxstats:md5,3db8f88c1f836eb5d10f37ca1df81ae9",
                "jurkat_REP2.sorted.bam:md5,f27e3736718fec53c189f7866932d28d",
                "jurkat_REP2.sorted.bam.bai:md5,a2f3e49a63c02bde49c24c56a1050606",
                "jurkat_REP2.stats:md5,f3be1f30d42067a85b0487bec047c49b"
            ]
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-02-13T22:00:59.731946823"
    },
    "Should run with bwamem2": {
        "content": [
            138,
            [
                "cd4_REP1.coverage.hist.txt:md5,7cbb473be8d3b32ff2e52fdf4e5d10d2",
                "cd4_REP1.coverage.stats.txt:md5,f1471b61ac17dba283d80e08450c7e55",
                "cd4_REP2.coverage.hist.txt:md5,8d5258a0882494bc4e3f1aa6aa5ed685",
                "cd4_REP2.coverage.stats.txt:md5,3cdb4473211f9da44166ffa6aaa5b602",
                "cd4_REP3.coverage.hist.txt:md5,4a8397c1ea08f35ca7046e3c2c014a08",
                "cd4_REP3.coverage.stats.txt:md5,b88592b469d504a93f2389255ac15f38",
                "cd4_REP4.coverage.hist.txt:md5,d2cbf89e1f753d99bcd03b4e472270dd",
                "cd4_REP4.coverage.stats.txt:md5,4c2b5fe6793aab8560cc87c176ac06ab",
                "jurkat_REP1.coverage.hist.txt:md5,71a893d9d1d55fd47399f6d47b628d6e",
                "jurkat_REP1.coverage.stats.txt:md5,381c69a30099d82066a959deab1a2569",
                "jurkat_REP2.coverage.hist.txt:md5,d124bdadaf2dc5cdd9e3db5a1ab4c588",
                "jurkat_REP2.coverage.stats.txt:md5,b1940346719e4070091f2eef6e508f49"
            ],
            [
                "cd4_intersect.saf:md5,149347636622d571c3c3ef762c1698ea"
            ],
            "cd4.bed:md5,ab94918610a560772fdbec591200295f",
            "jurkat.bed:md5,41908723fa423a78476feb2e7627a3a4",
            [
                "cd4_REP1.c_curve.txt:md5,cf4743abdd355595d6ec1fb3f38e66e5",
                "cd4_REP1.command.log:md5,d41d8cd98f00b204e9800998ecf8427e",
                "cd4_REP1.lc_extrap.txt:md5,6d0e7f782f34ee54213c022cbe19df5b",
                "cd4_REP2.c_curve.txt:md5,cf4743abdd355595d6ec1fb3f38e66e5",
                "cd4_REP2.command.log:md5,d41d8cd98f00b204e9800998ecf8427e",
                "cd4_REP2.lc_extrap.txt:md5,6aac6dd78dfd85a18fadd20be7377019",
                "cd4_REP3.c_curve.txt:md5,cf4743abdd355595d6ec1fb3f38e66e5",
                "cd4_REP3.command.log:md5,d41d8cd98f00b204e9800998ecf8427e",
                "cd4_REP3.lc_extrap.txt:md5,02c4b0fc0afd282a402a316d44598173",
                "cd4_REP4.c_curve.txt:md5,cf4743abdd355595d6ec1fb3f38e66e5",
                "cd4_REP4.command.log:md5,d41d8cd98f00b204e9800998ecf8427e",
                "cd4_REP4.lc_extrap.txt:md5,9efbab9d2aee791e7e9cae9237f2c500",
                "jurkat_REP1.c_curve.txt:md5,cf4743abdd355595d6ec1fb3f38e66e5",
                "jurkat_REP1.command.log:md5,d41d8cd98f00b204e9800998ecf8427e",
                "jurkat_REP1.lc_extrap.txt:md5,390386e428ce783b3a373a5f60ee912f",
                "jurkat_REP2.c_curve.txt:md5,cf4743abdd355595d6ec1fb3f38e66e5",
                "jurkat_REP2.command.log:md5,d41d8cd98f00b204e9800998ecf8427e",
                "jurkat_REP2.lc_extrap.txt:md5,cfdda91d83dd381dcefdc754e4b4c54c"
            ],
            [
                "GRCh38_chr21.fa.fai:md5,6ea51e1c1231da5e49212ea9f493d703",
                "cd4_REP1.flagstat:md5,863e2d506d5cc4239af98a5f31bbc906",
                "cd4_REP1.idxstats:md5,b1dd8bcbd23c53c21f0e11082d9315f2",
                "cd4_REP1.sorted.bam:md5,588def280611859a2b9c43c564975dc3",
                "cd4_REP1.sorted.bam.bai:md5,18d34e747d0986c043a6f0d14dee456b",
                "cd4_REP1.stats:md5,f4aa5b8d3fea2797724bf2fa12961036",
                "cd4_REP2.flagstat:md5,0fd86dbf8f799fad49ba471702979bdc",
                "cd4_REP2.idxstats:md5,53204e4c6a9f68664087e4a8123be46a",
                "cd4_REP2.sorted.bam:md5,0b67045362a25692019cb9fd57bb2df5",
                "cd4_REP2.sorted.bam.bai:md5,b03b5ced803ae927ab5a2f601abd3dcc",
                "cd4_REP2.stats:md5,9952636336505197c44c9ce1f7314096",
                "cd4_REP3.flagstat:md5,793c9cadbf80e35dbae3774e9b5bd93f",
                "cd4_REP3.idxstats:md5,42de227ff397013cf288759871b0e938",
                "cd4_REP3.sorted.bam:md5,7596ac9246a703dc8bce2cc1ae2b99d6",
                "cd4_REP3.sorted.bam.bai:md5,2a5802af6f299dd0f7ab8eee9552100a",
                "cd4_REP3.stats:md5,4f63629b6850326e1dbf449732bc7bf3",
                "cd4_REP4.flagstat:md5,ed6c609505097f3cc9d9902d69e016f8",
                "cd4_REP4.idxstats:md5,959dc185ae59de4cebe374026711a55e",
                "cd4_REP4.sorted.bam:md5,f63e2cdb10fa5b28acc564fb2f99ef3a",
                "cd4_REP4.sorted.bam.bai:md5,13d7de8a2c84b23998d59e41f78dc79c",
                "cd4_REP4.stats:md5,87c41cda4168ed794b640bfbd0e99d45",
                "jurkat_REP1.flagstat:md5,fd5f02b0f02a407447b804b1d80f5421",
                "jurkat_REP1.idxstats:md5,c61af0847c1ad76c06a8de2815975b32",
                "jurkat_REP1.sorted.bam:md5,f1d8db3612bf8917f723162f9ae07226",
                "jurkat_REP1.sorted.bam.bai:md5,19066b9272a3e73f9fc853be05f21220",
                "jurkat_REP1.stats:md5,677f4ec974fed98323ec41ab2f96049d",
                "jurkat_REP2.flagstat:md5,86ed47bd41a745ab59de473082c7742d",
                "jurkat_REP2.idxstats:md5,3db8f88c1f836eb5d10f37ca1df81ae9",
                "jurkat_REP2.sorted.bam:md5,a2c15ebbc6d24e0dabaa15b57a900248",
                "jurkat_REP2.sorted.bam.bai:md5,ae37c8fe3e780d068f5e32310bbcf387",
                "jurkat_REP2.stats:md5,f3be1f30d42067a85b0487bec047c49b"
            ]
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-02-13T22:02:42.148478009"
    },
    "Should run with dragmap": {
        "content": [
            138,
            [
                "cd4_REP1.coverage.hist.txt:md5,bff6862bc8e2015eef927b40671dec1b",
                "cd4_REP1.coverage.stats.txt:md5,c1e912cd0bebc51b289160ff7ab3365a",
                "cd4_REP2.coverage.hist.txt:md5,edadf440d1c3658118ffd6b6a46821f8",
                "cd4_REP2.coverage.stats.txt:md5,cb1ce057d05fc53e7ad9b0b2e3d5aaad",
                "cd4_REP3.coverage.hist.txt:md5,096ce6905c96587ba80ebcbe5893cf3e",
                "cd4_REP3.coverage.stats.txt:md5,f4b88f4eebab0c010a1cfdc3404dfafb",
                "cd4_REP4.coverage.hist.txt:md5,01f93b5c248e0b78f6d94e28eee3f9e7",
                "cd4_REP4.coverage.stats.txt:md5,a4c6f61be8263e7461f8b961cfcc9053",
                "jurkat_REP1.coverage.hist.txt:md5,468c864f655947edb3b306cb7a72633d",
                "jurkat_REP1.coverage.stats.txt:md5,338f3095395c8d8f172e0e27f9a5a60a",
                "jurkat_REP2.coverage.hist.txt:md5,03301957b2f29c43cb8ea3462998c315",
                "jurkat_REP2.coverage.stats.txt:md5,c3fc3f11e84f4d075ba95adbdbb2ee29"
            ],
            [
                "cd4_intersect.saf:md5,149347636622d571c3c3ef762c1698ea"
            ],
            "cd4.bed:md5,e9380700e48c1e52f76a2bb3fa4f59d9",
            "jurkat.bed:md5,fe88d4dc3c100262d7aec484e15b1c73",
            [
                "cd4_REP1.c_curve.txt:md5,cf4743abdd355595d6ec1fb3f38e66e5",
                "cd4_REP1.command.log:md5,d41d8cd98f00b204e9800998ecf8427e",
                "cd4_REP1.lc_extrap.txt:md5,5ecde39eb3bbec0b056b57a5f95f10e7",
                "cd4_REP2.c_curve.txt:md5,cf4743abdd355595d6ec1fb3f38e66e5",
                "cd4_REP2.command.log:md5,d41d8cd98f00b204e9800998ecf8427e",
                "cd4_REP2.lc_extrap.txt:md5,abb6a284d959110d35dd3b9659244188",
                "cd4_REP3.c_curve.txt:md5,cf4743abdd355595d6ec1fb3f38e66e5",
                "cd4_REP3.command.log:md5,d41d8cd98f00b204e9800998ecf8427e",
                "cd4_REP3.lc_extrap.txt:md5,af11536a40238dc3417468a2f670126c",
                "cd4_REP4.c_curve.txt:md5,cf4743abdd355595d6ec1fb3f38e66e5",
                "cd4_REP4.command.log:md5,d41d8cd98f00b204e9800998ecf8427e",
                "cd4_REP4.lc_extrap.txt:md5,b36b409ec88047cd354eb04a42b852e5",
                "jurkat_REP1.c_curve.txt:md5,cf4743abdd355595d6ec1fb3f38e66e5",
                "jurkat_REP1.command.log:md5,d41d8cd98f00b204e9800998ecf8427e",
                "jurkat_REP1.lc_extrap.txt:md5,024057a6211bf22384df658bce250d21",
                "jurkat_REP2.c_curve.txt:md5,cf4743abdd355595d6ec1fb3f38e66e5",
                "jurkat_REP2.command.log:md5,d41d8cd98f00b204e9800998ecf8427e",
                "jurkat_REP2.lc_extrap.txt:md5,33bdd91e7df616988c1ec043818b346b"
            ]
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2023-08-25T01:46:44+0000"
    }
}